### ISSUE TITLE
proposal: `useless_async`

### ISSUE THREAD

**Issue by DetachHead** [dart-lang/linter#3777]
## Description

warns when using the `async` keyword when the function doesn't need to

## Details

using the `async` keyword where it's not needed can lead to confusion and misunderstandings of how `Futures` and `async`/`await` works.

eslint has a rule for this called [`require-await`](https://eslint.org/docs/latest/rules/require-await)

## Kind

from the eslint rule docs:

> Asynchronous functions that don’t use `await` might not need to be asynchronous functions and could be the unintentional result of refactoring.

## Good Examples 

```dart
Future<String> getValue() async {
  // ...
}

Future<String> foo() {
  return getValue();
}
```

## Bad Examples

i often see code like this in real life. and when pointing it out it's usually due to a lack of understanding of what exactly `async` does.

```dart
Future<String> getValue() async {
  // ...
}

Future<String> foo() async { // this `async` is not necessary
  return getValue();
}
```

### Discussion checklist

- [X] List any existing rules this proposal modifies, complements, overlaps or conflicts with.
- [X] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [X] If there's any prior art (e.g., in other linters), please add references here.
- [X] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn't any corresponding advice, should there be?)
- [X] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

[Writing Lints]: https://github.com/dart-lang/linter/blob/main/doc/writing-lints.md
[Effective Dart]: https://dart.dev/guides/language/effective-dart
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[SDK Tracker]: https://github.com/dart-lang/sdk/issues

**Comment by incendial** [comment]
@DetachHead have you tried https://dartcodemetrics.dev/docs/rules/common/avoid-redundant-async? 🙂

**Comment by DetachHead** [comment]
@incendial thanks! looks like that rule was added since i last looked at `dart_code_metrics`

**Comment by incendial** [comment]
@DetachHead hm, maybe you have suggestions how we can help discover new rules?

**Comment by DetachHead** [comment]
personally i like to set linters to be as strict as possible. that way, i can see the errors in my codebase and decide for myself easily which new rules i want to disable. maybe if there was an option to enable all rules by default and have a list of rules to explicitly disable

**Comment by incendial** [comment]
You can, DCM now supports presets https://dartcodemetrics.dev/docs/getting-started/configuration#extending-an-existing-configuration-preset and https://github.com/dart-code-checker/dart-code-metrics/tree/master/lib/presets, this should cover you use-case

**Comment by DetachHead** [comment]
awesome, thanks!
